packages/opentelemetry-types/src/trace/tracer.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span } from './span';
+import { Attributes } from './attributes';
+import { SpanKind } from './span_kind';
+import { SpanContext } from './span_context';
+
+/**
+ * Options needed for span creation
+ *
+ * @todo: Move into module of its own
+ */
+export interface SpanOptions {
+  /**
+   * The SpanKind of a span
+   */
+  kind?: SpanKind;
+
+  /**
+   * A spans attributes
+   */
+  attributes?: Attributes;
+
+  /**
+   * Indicates that events are being recorded for a span
+   */
+  isRecordingEvents?: boolean;
+
+  /**
+   * The parent span
+   */
+  parent?: Span | SpanContext;
+
+  /**
+   * The start timestamp of a span
+   */
+  startTime?: number;
+}
+
+/**
+ * Tracer provides an interface for creating spans and propagating context in-process.
+ */
+export interface Tracer {
+  /**
+   * Returns the current Span from the current context if available.
+   *
+   * If there is no Span associated with the current context, a default Span
+   * with invalid SpanContext is returned.
+   *
+   * @returns Span The currently active Span
+   */
+  getCurrentSpan(): Span;
+
+  /**
+   *
+   * @param name The name of the span
+   * @param [options] SpanOptions used for span creation
+   * @returns Span The newly created span
+   */
+  start(name: string, options?: SpanOptions): Span;
```

Comment:
As per the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/tracing-api.md#startspan), the name should be `startSpan` instead of just `start`.